doc/sphinx/src/models.rst

```diff
@@ -3,3 +3,409 @@
 EOS Models
 ===========
 
+The mathematical descriptions of these models are presented below while the
+details of using them is presented in the description of the 
+:doc:`EOS API <using-eos>`.
+
+EOS Primer
+----------
+
+An equation of state (EOS) is a constituitive model that generally relates
+thermodynamic quantities such as pressure, temperature, density, internal
+energy, free energy, and entropy consistent with the constraints of equillibrium
+thermodynamics.
+
+``singularity-eos`` contains a number of equations of state that are most useful
+in hydrodynamics codes. All of the equations of state presented here are
+considered "complete" equations of state in the sense that they contain all the
+information needed to derive a given thermodynamic quantity. However, not all
+variables are exposed since the emphasis is on only those that are needed for
+hydrodynamics codes. An incomplete equation of state is often sufficient for
+pure hydrodynamics since it can relate pressure to a given energy-density state,
+but it is missing information that relates the energy to temperature (which in
+turn provides a means to calculate entropy).
```

Comment:
Beyond the scope of this PR, but at some point I'd like to attack entropy more formally. Having access to entropy vs rho and T (for example) would be a generically useful capability. It's needed, actually, for some applications.

Reply:
For most equations of state it will be a simple add as well. I'm happy to code up the actual isentropes if you create all the plumbing.

Reply:
It's on my to do list, as it's needed for my NGP work. :D